### DISCUSSION TITLE
Add "diff" feature to Response History

### DISCUSSION THREAD

**Discussion by JackBeQuick87** [Kong/insomnia#8115] [Ideas]
Please implement a feature that will users to compare ("diff") two responses.  The user should be able to select which responses on which they would like have the diff comparison performed.  A great subfeature would be to have an option to automatically update the comparison between the most recent response and the second-most recent response, or a selected fixed response from the history.

This is the "Response history" referenced in my request.
https://docs.insomnia.rest/insomnia/responses#access-response-history

Note: This request is similar to #3552, but that request had multiple aspects.  I do not feel that my request is an extension of that issue.

**Comment by zeatful** [comment]
I have actually implemented something similarly locally and have been testing it.

Would the contributors be willing to consider this if I submit this PR for acceptance?

**Reply by zeatful** [reply]
if not I can look at making it a plugin because I'm already getting tons of value out of my local changes (to the fact I'm using my local version over the official release)

**Reply by zeatful** [reply]
<img width="1415" height="978" alt="Screenshot 2025-10-25 at 3 03 17 PM" src="https://github.com/user-attachments/assets/8aeddc61-90fa-4582-bbbf-1ac6af690722" />
Essentially you use the runner, and under advanced, you pick `enable environment comparison` and you can specify separate environments to essentially run a regression to quickly diff responses maybe from your local env to a dev or other api url.
